src/realm/util/file.cpp

```diff
@@ -198,6 +208,18 @@ void remove_dir_recursive(const std::string& path)
 }
 
 
+bool try_remove_dir_recursive(const std::string& path)
+{
+    try {
+        remove_dir_recursive(path);
+    }
+    catch (const File::NotFound&) {
+        return false;
+    }
+    return true;
+}
```

Comment:
@kspangsege Thanks for reviewing.
I agree with the race conditions.
I implicitly assumed that the functions only promised the specified behavior
if no other threads interfered.

I should document it. Agree.

Also, an application probably has lots of problems if one thread creates files in the directory while another is removing it.